scripts/client_test/cli_test.py

```diff
@@ -27,233 +84,286 @@ class TestCli:
     runtime = Runtime()
     evaluation = Evaluation()
 
-    def __init__(self, work_dir: str) -> None:
+    def __init__(
+        self,
+        work_dir: str,
+        thread_pool: ThreadPoolExecutor,
+        server_url: t.Optional[str],
+        server_project: str = "starwhale",
+    ) -> None:
         self._work_dir = work_dir
+        self.executor = thread_pool
+        self.server_url = server_url
+        self.server_project = server_project
+        self.datasets: t.Dict[str, t.List[URI]] = {}
+        self.runtimes: t.Dict[str, URI] = {}
+        self.models: t.Dict[str, URI] = {}
+        if self.server_url:
+            logger.info(f"login to server {self.server_url} ...")
+            assert self.instance.login(url=self.server_url)
+
+    def build_dataset(self, _workdir: str, handler: str = "") -> t.Any:
+        self.select_local_instance()
+        _uri = Dataset.build_with_api(workdir=_workdir, handler=handler)
+        if self.server_url:
+            assert self.dataset.copy(
+                src_uri=_uri.full_uri,
+                target_project=f"cloud://server/project/{self.server_project}",
+                force=True,
+            )
+        dss_ = self.datasets.get(_uri.object.name, [])
+        dss_.append(_uri)
+        self.datasets.update({_uri.object.name: dss_})
+        assert len(self.dataset.list())
+        assert self.dataset.info(_uri.full_uri)
+        return _uri
+
+    def build_model(
+        self,
+        _workdir: str,
+    ) -> t.Any:
+        self.select_local_instance()
+        _uri = Model.build_with_api(workdir=_workdir)
+        if self.server_url:
+            assert self.model.copy(
```

Comment:
`self.model`  typo? I only found `self.models` definition in the init function.

Reply:
line134-135 has the same issue.

Reply:
These are class variables 
https://github.com/star-whale/starwhale/blob/cbe7698e719bdc454cec16f165362295a457876f/scripts/client_test/cli_test.py#L79-L85

Reply:
`self.model` and `self.models` have similar meanings. could we use another accurate word?

Reply:
sure, done
https://github.com/star-whale/starwhale/blob/4590c633bf1c80dadc0d1d4c49bae64e5351a352/scripts/client_test/cli_test.py#L79-L85